### ISSUE TITLE
on ubuntu apt install httpie install v1.0.3

### ISSUE THREAD

**Issue by paullaffitte** [httpie/httpie#1003]
Installing httpie on ubuntu 20.04 with `apt install httpie` will install httpie 1.0.3

**Comment by jakubroztocil** [comment]
Thanks for the report. We’re looking into having the Ubuntu package updated.

**Comment by sustefil** [comment]
+1

There is no `https` in version 1.0.3, which makes it quite unusable on Ubuntu20

**Comment by mtwolak** [comment]
As a workaround you can install v.2 by grabbing this version from [Debian's site](https://packages.debian.org/sid/all/httpie/download).

**Comment by 1Nick1Name** [comment]
Now almost a year later, any progress on this issue?

**Comment by BoboTiG** [comment]
Ubuntu is shipping `snap`, you could install the latest version using it (here is the [package](https://snapcraft.io/httpie)):
```bash
sudo snap instal httpie
```
It is the preferred way to install it on Ubuntu.

We may improve the situation using `apk` but that takes time and it will not be available on all versions on Ubuntu. Have a look at https://repology.org/project/httpie/versions where you can find which version is available for your OS.

**Comment by jrtitus** [comment]
@BoboTiG Is there supposed to be a `latest/stable` when installing via `snap`?

`snap info httpie` returns

```
channels:
  latest/stable:    –                         
  latest/candidate: –                         
  latest/beta:      –                         
  latest/edge:      2.5.0 2021-09-12 (25) 6MB -
installed:          2.5.0            (25) 6MB -
```

I needed to do `snap install --edge httpie`

**Comment by somera** [comment]
I took the pip installation.

**Comment by BoboTiG** [comment]
@jrtitus the stable version will come later. I knew `snap` would propose the version from edge, it is expected. Tomorrow, 2.5.0 will become stable and snap will handle it perfectly.

I would like some feedbacks before moving the package to stable ;)

**Comment by BoboTiG** [comment]
@somera wise move 👍

**Comment by 1Nick1Name** [comment]
To my other dismay, does not work on in Ubuntu on wsl.

Mickaël Schoentgen ***@***.***> schrieb am So., 12. Sept.
2021, 22:20:

**Comment by BoboTiG** [comment]
> To my other dismay, does not work on in Ubuntu on wsl.

Could you open a specific issue with more context an details please? I am afraid that WSL is another beast on its own.

**Comment by 1Nick1Name** [comment]
> 
> 
> > To my other dismay, does not work on in Ubuntu on wsl.
> 
> @1Nick1Name, could you open a specific issue with more context an details please? I am afraid that WSL is another beast on its own.

Sorry, what I meant to write was that snap does not work on Ubuntu on wsl. So using snap is not a viable alternative to apt.

**Comment by somera** [comment]
> > > To my other dismay, does not work on in Ubuntu on wsl.
> > 
> > 
> > @1Nick1Name, could you open a specific issue with more context an details please? I am afraid that WSL is another beast on its own.
> 
> Sorry, what I meant to write was that snap does not work on Ubuntu on wsl. So using snap is not a viable alternative to apt.

But this

`pip install --user --upgrade httpie`

works on wsl.

**Comment by BoboTiG** [comment]
> Sorry, what I meant to write was that snap does not work on Ubuntu on wsl. So using snap is not a viable alternative to apt.

@1Nick1Name, indeed, [Snap documentation](https://snapcraft.io/docs/build-on-windows) states:

> While it is possible to build, publish and release snaps using WSL, it’s not currently possible to install or run them in that environment.

So let's go with `pip`, it will always work.

**Comment by 1Nick1Name** [comment]
> > > > To my other dismay, does not work on in Ubuntu on wsl.
> > > @1Nick1Name, could you open a specific issue with more context an details please? I am afraid that WSL is another beast on its own.
> > Sorry, what I meant to write was that snap does not work on Ubuntu on wsl. So using snap is not a viable alternative to apt.
> But this
> `pip install --user --upgrade httpie`
> works on wsl.

Yea maybe. At the moment I am trying to limit the amount of rabbit holes I jump into to have a good chase. So I'd prefer to stay with the Ubuntu packet manager.
That might be kind of the point to my comment and maybe the OPs post. We all have limited time, so there is no good reason to concern yourselves with a zoo of possible alternatives. Either the httpie community wants to support Ubuntu then it might be a good idea to actually provide an apt packet that is not antediluvian or the community does not wish to support Ubuntu, that's OK too, than I'll be a bit sad but I'll have a chance to move on and keep using the good old curl. What I deem to be a bit unpleasant is: saying the package will be updated and than keep it hanging.

**Comment by BoboTiG** [comment]
@1Nick1Name I understand your frustration and get it. We are actively looking to improve the situation.

**Comment by isidentical** [comment]
From what I can reproduce, the latest version available in the ubuntu repos at the moment is 2.4.0. Our debian package is now 2.6 on the Debian Sid environment, so we are hopeful that the next version of ubuntu will install either 2.6.0 or 3.0.0 (depending on whether we'll release it or not). We are hoping that in the future, the delays between package managers are going to as noticeable as this one. Closing as it is resolved for our end.
